client/src/test/java/com/phaxio/unittests/AccountTest.java

```
@@ -0,0 +1,25 @@
+package com.phaxio.unittests;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.phaxio.entities.Account;
+import com.phaxio.helpers.Responses;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+
+public class AccountTest {
+    @Test
+    public void ignoresExtraFields () throws IOException {
```

in the interest of not having to update these tests for future versions of the API, i think we should simply assert that readValue is not throwing UnexpectedPropertyException instead of assertEqualsing all the properties for what the current lib expects in the current version of the API it's mapped to.  that's the main focus of these tests anyway.

---

let me clarify - this specific test should not have to be updated.  but now that you've done the work of enumerating all the fields, perhaps we should move that code into another test that verifies the resulting data.

---

I originally thought the same thing as your first comment - I should just try to deserialize an object and if it works, it works. I then decided that I don't just care if it ignores certain fields, it also needs to pay attention to other fields.

The other thing is that JUnit doesn't have a clear way of communicating that we don't expect an exception - there's no AssertDoesntThrow. The tests would be deserialization and then just end - a bit anticlimactic. We could perhaps rename the tests to `deserializesSpecifiedFieldsOnly` or `deserializesIgnoringExtraFields`.

If you want it broken out, though, that's up to you, I just wanted to present my reasoning.

---

To address the concern about the API changing: we need to verify somewhere that the JSON deserialization is working somewhere and since we're hand-specifying the field mapping, we have to verify each field.

We could however go to an automatic conversion strategy: the lib we're using allows us to specify that a class uses conventional CamelCase and the JSON uses snake_case. I believe I specified the fields by hand originally because I was going to use more conventional Java naming before you specified that you wanted the exact names just converted to CamelCase.

Regardless of whether we use an automatic mapping strategy or not, I still think we need tests on individual fields because we could make a mistake naming the Java class fields. This is especially true now that we're silently ignoring fields (whereas before, we'd get an exception if a field was in the JSON but not mapped which would've detected these types of errors).

---

Yeah, agreed re needing to test the deserialization with the way things are currently happening.  I'm just thinking that the deserialization part should be tested in a different test than the "ignoring properties" test, as its really testing 2 different things.  For now I think it's fine, and we can separate that at a later date.  We may want to switch to automatic serialization, but as it stands this is enough to satisfy the feature.